### ISSUE TITLE
fails to build against tinyxml2 5.0.0

### ISSUE THREAD

**Issue by wjwwood** [eProsima/Fast-RTPS#113]
I was seeing about upgrading our macOS build machines to the latest versions of software available from Homebrew, and one of the new things is `tinyxml2` version `5.0.0`. However, that seems to make Fast-RTPS not compile:

```
[100%] Linking CXX shared library libfastrtps.dylib
Undefined symbols for architecture x86_64:
  "tinyxml2::XMLDocument::LoadFile(char const*)", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) in EDPStaticXML.cpp.o
  "tinyxml2::XMLDocument::XMLDocument(bool, tinyxml2::Whitespace)", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) in EDPStaticXML.cpp.o
  "tinyxml2::XMLDocument::~XMLDocument()", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) in EDPStaticXML.cpp.o
  "tinyxml2::XMLElement::FindAttribute(char const*) const", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLReaderEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLWriterEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
  "tinyxml2::XMLElement::GetText() const", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLParticipantEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLReaderEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLWriterEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
  "tinyxml2::XMLElement::Attribute(char const*, char const*) const", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLReaderEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLWriterEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
  "tinyxml2::XMLAttribute::QueryIntValue(int*) const", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLReaderEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLWriterEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
  "tinyxml2::XMLNode::FirstChildElement(char const*) const", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLParticipantEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLReaderEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLWriterEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
  "tinyxml2::XMLNode::NextSiblingElement(char const*) const", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLParticipantEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLReaderEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLWriterEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
  "tinyxml2::XMLNode::Value() const", referenced from:
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLParticipantEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLReaderEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
      eprosima::fastrtps::rtps::EDPStaticXML::loadXMLWriterEndpoint(tinyxml2::XMLElement*, eprosima::fastrtps::rtps::StaticRTPSParticipantInfo*) in EDPStaticXML.cpp.o
ld: symbol(s) not found for architecture x86_64
```

-- http://ci.ros2.org/job/ci_osx_testing/2/console

I'll probably spend a few minutes to see if I can get a pull request open to fix this, but that depends on the extent of the changes between 5.0.0 and the previous version we were using, `4.0.1`. I'm hopeful the fix is easy given that it compiles but just fails to link. That means to me that it probably just needs a CMake fix. Or that something else fishy is going on.

Here's the change on the macOS build machine:

```
dosa:~ osrf$ brew cleanup tinyxml2
Removing: /usr/local/Cellar/tinyxml2/4.0.1... (7 files, 137.2KB)
==> This operation has freed approximately 137.2KB of disk space.
dosa:~ osrf$ brew info tinyxml2
tinyxml2: stable 5.0.0 (bottled), HEAD
Improved tinyxml (in memory efficiency and size)
http://grinninglizard.com/tinyxml2
/usr/local/Cellar/tinyxml2/5.0.0 (11 files, 148.7KB) *
  Poured from bottle on 2017-06-22 at 17:12:05
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/tinyxml2.rb
==> Dependencies
Build: cmake ✔
```

This does not need to block the upcoming 1.5.0 release (in my opinion), as I can simply roll back the version on our machines until this is resolved.

**Comment by richiware** [comment]
@wjwwood Do you still have problems with TinyXML2 5.0.0 after merging the PR? If not we can close this issue. Thanks

**Comment by bruceSz** [comment]
Any updates on this thread? 
I met same issue on Mac

**Comment by rgreid** [comment]
It seems Brew has upgraded tinyxml2 to 7.0.1

A temporary workaround is to install and pin the tinyxml2 version:
```
brew uninstall --force tinyxml2
brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/0b3ee2f67043af1b4270096c13350420e3290399/Formula/tinyxml2.rb
brew pin tinyxml2
```

**Comment by MiguelCompany** [comment]
Please see PR #349 which should fix the issue with tinyxml2 >= 7.0.0
